element.h

```diff
@@ -12,8 +21,10 @@ enum Element {
   ALL=Element::FIRE | Element::WATER | Element::EARTH | Element::SPIRIT | Element::LIGHTNING,
 };
 
+// contains all elements being used in the game.
 const std::vector<Element> ELEMENTLIST = {Element::FIRE, Element::WATER, Element::LIGHTNING, Element::EARTH, Element::SPIRIT};
 
-bool elementInSearch(const Element, const int);
+// checks if an element is in a given search mask.
```

Comment:
I can see any information in this comment I couldn't figure out from a well named function. I should be able to figure this out from the name (the name probably needs a bit of work too, but that's not for this PR).